src/sparseml/recipe_template/utils.py

```diff
@@ -82,6 +82,37 @@ def modifier_recipe_variables(self) -> Dict[str, Any]:
         """
         return self.__modifier_recipe_variables
 
+    @staticmethod
+    def build_recipe_from_modifier_info(
+        modifier_info_groups: Dict[str, List["ModifierBuildInfo"]],
+        convert_to_md: bool = False,
+    ) -> str:
+        """
+        :param modifier_info_groups: A dict mapping b/w modifier group names
+            and List[ModifierBuildInfo] representing information for each modifier
+            in the group
+        :param convert_to_md: A boolean flag, if True the string is returned with
+            yaml front matter that can be embedded in a .md file
+        :return: A yaml or md string representing a valid recipe
+        """
+        recipe_variables = {}
+        modifier_groups = defaultdict(list)
+
+        for group_name, modifier_group in modifier_info_groups.items():
+            for modifier_info in modifier_group:
+                if modifier_info is not None:
+                    recipe_variables.update(modifier_info.modifier_recipe_variables)
+                    modifier_groups[group_name].append(modifier_info.modifier_builder)
```

Comment:
So it seems like the whole reason you need the builder is to contain these two things:
1. The recipe variables
2. The builder, which is just a `ModifierYAMLBuilder(modifier)`

Do you think we could just pass around `Tuple[Modifier, Dict[str, Any]]` instead of adding ModifierBuilderInfo?